src/ota_stm32f.c

```diff
@@ -0,0 +1,228 @@
+#include "flash.h"
+#include "log.h"
+#include "ota.h"
+
+#if MG_OTA == MG_OTA_STM32F
+
+static bool mg_stm32f_write(void *, const void *, size_t);
+static bool mg_stm32f_swap(void);
+
+static struct mg_flash s_mg_flash_stm32f = {
+    (void *) 0x08000000,  // Start
+    0,                    // Size, FLASH_SIZE_REG
+    0,                    // Irregular sector size
+    32,                   // Align, 256 bit
+    mg_stm32f_write,
+    mg_stm32f_swap,
+};
+
+#define FLASH_BASE 0x40023c00  // Base address for bank1
+#define FLASH_KEYR 0x04
+#define FLASH_SR 0x0c
+#define FLASH_CR 0x10
+#define FLASH_OPTCR 0x14
+#define FLASH_SIZE_REG_F7 0x1FF0F442
+#define FLASH_SIZE_REG_F4 0x1FFF7A22
+
+#define STM_DBGMCU_IDCODE 0xE0042000
+#define STM_DEV_ID (MG_REG(STM_DBGMCU_IDCODE) & (MG_BIT(12) - 1))
+#define SYSCFG_MEMRMP 0x40013800
+
+static uintptr_t flash_size_reg_location() {
+  if (STM_DEV_ID >= 0x449) {
+    return FLASH_SIZE_REG_F7;
+  } else {
+    return FLASH_SIZE_REG_F4;
+  }
+}
+
+static size_t flash_size(void) {
+  uintptr_t flash_size_reg = flash_size_reg_location();
+  return (MG_REG(flash_size_reg) & 0xFFFF) * 1024;
+}
+
+MG_IRAM int flash_dual_bank(void) {
```

Comment:
We don't need this public one anymore.
I would go for `MG_IRAM static bool is_dualbank(void)` + `MG_IRAM static uint32_t flash_bank(void *addr)` as in the H7
https://github.com/cesanta/mongoose/blob/7d336991739ce0f53a43b51aeb829a437177201a/src/ota_stm32h7.c#L38-L40
https://github.com/cesanta/mongoose/blob/7d336991739ce0f53a43b51aeb829a437177201a/src/ota_stm32h7.c#L81-L86